### ISSUE TITLE
`_GenericAlias.__mro__` does not error

### ISSUE THREAD

**Issue by NCPlayz** [microsoft/pyright#6244]
**Describe the bug**
While investigating python/cpython#111353, I encountered a bug with pyright where it validates `__mro__` on a `_GenericAlias`, which at runtime is not accessible. 

**Code or Screenshots**
```python
from typing import Generic, TypeVar

T = TypeVar('T')

class Foo(Generic[T]):
    ...

Foo[str].__mro__
```

**VS Code extension or command-line**
pyright 1.1.333

**Comment by erictraut** [comment]
There is no type error in this code, so pyright doesn't generate an error. `_GenericAlias` is an internal implementation detail of the runtime, and the fact that the runtime doesn't support an `__mro__` attribute is more of a runtime bug, IMO. Based on the type information provided to pyright, this code is fine.

For comparison, mypy also doesn't generate an error in this case.

It would be theoretically possible to add special-case knowledge within pyright to detect and report this particular condition, but we generally don't do this unless it's a frequent source of bugs in code. I don't think this qualifies. We could revisit this in the future if we receive additional signal from pyright users.

**Comment by NCPlayz** [comment]
Thanks. I agree to be honest, and I wasn't sure if I wanted to report this as a bug or say that typecheckers  follow this pattern (regarding the issue).
